modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldSettingServiceTest.java

```diff
@@ -0,0 +1,296 @@
+			String message = principalException.getMessage();
+
+			Assert.assertTrue(
+				message.contains(
+					"User " + _defaultUser.getUserId() +
+						" must have UPDATE permission for"));
+		}
+
+		_testAddObjectFieldSetting(_objectField.getObjectFieldId(), _user);
+	}
+
+	@Test
+	public void testDeleteObjectFieldSetting() throws Exception {
+		try {
+			_testDeleteObjectFieldSetting(_defaultUser);
+
+			Assert.fail();
+		}
+		catch (PrincipalException.MustHavePermission principalException) {
+			String message = principalException.getMessage();
+
+			Assert.assertTrue(
+				message.contains(
+					"User " + _defaultUser.getUserId() +
+						" must have UPDATE permission for"));
+		}
+
+		_testDeleteObjectFieldSetting(_user);
+	}
+
+	@Test
+	public void testGetObjectFieldSetting() throws Exception {
+		try {
+			_testGetObjectFieldSetting(_defaultUser);
+		}
+		catch (PrincipalException.MustHavePermission principalException) {
+			String message = principalException.getMessage();
+
+			Assert.assertTrue(
+				message.contains(
+					"User " + _defaultUser.getUserId() +
+						" must have VIEW permission for"));
+		}
+
+		_testGetObjectFieldSetting(_user);
+	}
+
+	@Test
+	public void testUpdateObjectFieldSetting() throws Exception {
+		try {
+			_testUpdateObjectFieldSetting(_defaultUser);
+
+			Assert.fail();
+		}
+		catch (PrincipalException.MustHavePermission principalException) {
+			String message = principalException.getMessage();
+
+			Assert.assertTrue(
+				message.contains(
+					"User " + _defaultUser.getUserId() +
+						" must have UPDATE permission for"));
+		}
+
+		_testUpdateObjectFieldSetting(_user);
+	}
+
+	private ObjectFieldSetting _addObjectFieldSetting(User user)
+		throws Exception {
+
+		return _objectFieldSettingLocalService.addObjectFieldSetting(
+			user.getUserId(), _objectField.getObjectFieldId(),
+			StringUtil.randomId(), RandomTestUtil.randomBoolean(),
+			RandomTestUtil.randomString());
+	}
+
+	private void _setUser(User user) {
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(user));
+
+		PrincipalThreadLocal.setName(user.getUserId());
+	}
+
+	private void _testAddObjectFieldSetting(long objectFieldId, User user)
+		throws Exception {
+
+		ObjectFieldSetting objectFieldSetting = null;
+
+		try {
+			_setUser(user);
+
+			objectFieldSetting =
+				_objectFieldSettingService.addObjectFieldSetting(
+					objectFieldId, StringUtil.randomId(),
+					RandomTestUtil.randomBoolean(),
+					RandomTestUtil.randomString());
+		}
+		finally {
+			if (objectFieldSetting != null) {
+				_objectFieldSettingLocalService.deleteObjectFieldSetting(
+					objectFieldSetting);
+			}
+		}
+	}
+
+	private void _testDeleteObjectFieldSetting(User user) throws Exception {
+		ObjectFieldSetting deleteObjectFieldSetting = null;
+		ObjectFieldSetting objectFieldSetting = null;
+
+		try {
+			_setUser(user);
+
+			objectFieldSetting = _addObjectFieldSetting(user);
+
+			deleteObjectFieldSetting =
```

Comment:
Isn't `objectFieldSetting == deleteObjectFieldSetting`?

Reply:
Got it. Then can we change its name to `deletedObjectFieldSetting`? Since "delete" is a verb, it implies that it's a method, not an object.